frontend/src/metabase/components/SegmentedControl/SegmentedControl.tsx

```diff
@@ -21,35 +21,47 @@ export const optionShape = PropTypes.shape({
   selectedColor: PropTypes.string,
 });
 
-const propTypes = {
-  name: PropTypes.string,
-  value: PropTypes.any,
-  options: PropTypes.arrayOf(optionShape).isRequired,
-  variant: PropTypes.oneOf(["fill-text", "fill-background", "fill-all"]),
-  inactiveColor: PropTypes.string,
-  onChange: PropTypes.func,
-  fullWidth: PropTypes.bool,
+export type SegmentedControlOption = {
+  name?: React.ReactNode;
+  value: any;
```

Comment:
Can it be changed to a generic type?

Reply:
@ranquild good point, thanks! Mind taking another look once again?